libexec/rtld-elf/aarch64/rtld_start.S

```diff
@@ -119,25 +119,7 @@ END(.rtld_start)
  */
 ENTRY(C18N_SYM(_rtld_bind_start))
 #ifdef CHERI_LIB_C18N
-	/*
-	 * Get the caller's current stack top.
-	 */
-	get_untrusted_stk	c10
-	/*
-	 * Update the stack lookup table with the caller's current stack top and
-	 * get the caller's old stack top.
-	 */
-	update_stk_table	c11, c10, w12
-	/*
-	 * Switch to RTLD's stack.
-	 */
-	get_rtld_stk		c12
-	set_untrusted_stk	c12
-	/*
-	 * Save caller's old stack top.
-	 */
-	str	c11, [csp, #-CAP_WIDTH]!
-	mov	c17, c10
+	get_untrusted_stk	c17
```

Comment:
Doesn't everything now need to use c17?

Reply:
Or is csp here actually already the right stack to be using?

Reply:
(Or alternatively should this code run in restricted mode so it uses the same stack as the caller?)

Reply:
Under the purecap ABI, csp is the Executive stack. And under the benchmark ABI, csp is the caller's stack.

In both cases there's no security problem to temporarily push the caller's arguments on csp, I think.

Reply:
Then why are we even setting c17 here? AFAICT it's not used, the only reference is on the return path for the PLT-saved CLR, which has its own get_untrusted_stk just before it.

Reply:
c17 is used down below in `ldr PTR(2), [PTR(17), #0]`

Reply:
Oh I see, PTR obfuscated my search. I still think it would be simpler though if we just ran this code in restricted mode (for non-benchmark), then csp is always the caller's stack.

Reply:
(and with sub-library compartmentalised rtld that's how you'd probably model it, _rtld_bind in its own unprivileged compartment and then everything automatically works)

Reply:
Right. I've made all RTLD hooks that are inserted in GOTs run in restricted mode.